Protocol.md

```diff
@@ -23,188 +44,182 @@ The following figure gives a brief overview of the protocol:
     |       |                    |  (4) broadcast fund-tx
     |       |                    |       |
     +-------+                    +-------+
+    
+        - where node A is 'offerer' and node B is 'accepter'
 
-### Offer
-
-The initiating party starts the protocol by sending an `offer` message to the other party.
-This message includes the following information:
-1. Contract information (a map defined below)
-1. Oracle information (optional)
-1. A's fund amount (satoshi)
-1. A's public keys
-1. A's addresses
-1. A's funding inputs
-1. Estimatesmartfee (satoshi/vbyte)
-1. CET CSV delay
-1. CET locktime (corresponding to the contract maturity date)
-1. Refund locktime
-
-#### Contract information
-Contract information consists of a total value (sum of both outputs) and a map to be used to create CETs.
-The map consists of:
-`message` - `a_output` (satoshi)
-where `message` is the data on which the oracle will be providing a (set of) signatures, and `a_output` represents the amount that A will receive for this outcome (B's outcome can be computed by substracting `a_output` to the total value).
-The interpretation of the message semantics is left to the application.
+If this fails at any stage, or if one node decides the contract terms
+offered by the other node are not suitable, the contract negotiation
+fails.
 
-#### Oracle information
-The oracle public key and R point(s) to use to build the CETs as well as meta information to identify the oracle to be used in the contract.
-Example:
-`public_key` - `R_point`<sub>1</sub> - `R_point`<sub>2</sub> - `meta data`
-The meta data can either be human readable for a user to be able to interpret it, or machine readable so that an application can interpret it and present it to the user.
-If both parties already have this information, transmission is unnecessary.
+Note that multiple contracts can be open in parallel, as all DLC
+messages are identified by either a `temporary_contract_id` (before the
+funding transaction is created) or a `contract_id` (derived from the
+funding transaction).
+
+### The `offer_dlc` Message
+
+This message contains information about a node and indicates its
+desire to open a new contract. This is the first step toward creating
+the funding transaction and CETs.
+
+1. type: ??? (`offer_dlc`)
+2. data:
+   * [`byte`:`contract_flags`]
+   * [`chain_hash`:`chain_hash`]
+   * [`contract_info`:`contract_info`]
+   * [`oracle_info`:`oracle_info`]
+   * [`point`:`funding_pubkey`]
```

Comment:
I'm guessing `offer_dlc` assumes counterparty has already been established (whether off-chain or through an orderbook) since counterparty pubkey would be required to derive `funding_pubkey`

Reply:
I'm not sure what you mean by

> counterparty pubkey is required to derive `funding_pubkey`

as I don't believe this is true.

Regardless though, yes this entire document assumes the existence of some kind of communication layer as is stated at the top of the Messaging doc

Reply:
`funding_pubkey` is a 2-of-2 multisig Script right? So you need pubkey of both alice and bob to derive `funding_pubkey`

Reply:
`funding_pubkey` is just one party's public key (which is one of the two that goes into the funding script public key)